ValveResourceFormat/Resource/ResourceTypes/KV3.cs

```diff
@@ -0,0 +1,592 @@
+/*
+ * KV3Reader class.
+ * This class reads in valve KV3 files and stores them in its datastructure.
+ *  
+ * Interface:
+ *  KVFile file = KV3Reader.ParseKVFile( fileName );
+ *  String fileString = file.Serialize();
+ *  
+ * TODO:
```

Comment:
Nit: it's unnecessary to add TODOs in comments like this, either mention them in the pull request, or create separate issues after its merged.